File: frontend/src/types/security-agent.ts
export interface SecurityFindingItem {
  id: string;
  category: 'SQL_INJECTION' | 'SECRETS' | 'XSS' | 'JWT' | 'CSRF' | 'DEPENDENCY' | 'OWASP';
  owasp_category: string;
  cwe_id: string;
  title: string;
  severity: 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW';
  cvss_score: number;
  cvss_vector: string;
  file_path: string;
  line_number: number;
  code_snippet: string;
  recommendation: string;
  patch_diff?: string;
}

export interface SecurityAgentReportResponse {
  id: string;
  repository_id: string;
  risk_score: number;
  risk_level: 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW';
  critical_count: number;
  high_count: number;
  medium_count: number;
  low_count: number;
  findings: SecurityFindingItem[];
  owasp_distribution: Record<string, number>;
  chart_dataset: {
    severity_counts: Record<string, number>;
    category_breakdown: Array<{ name: string; count: number; color: string }>;
    cvss_trend: Array<{ label: string; score: number }>;
  };
  scanned_at: string;
}

File: backend/app/schemas/security_agent.py
import uuid
from datetime import datetime
from typing import Dict, List, Optional, Any
from pydantic import BaseModel

class SecurityFindingItem(BaseModel):
    id: str
    category: str # 'SQL_INJECTION' | 'SECRETS' | 'XSS' | 'JWT' | 'CSRF' | 'DEPENDENCY' | 'OWASP'
    owasp_category: str # e.g. 'A03:2021-Injection'
    cwe_id: str # e.g. 'CWE-89'
    title: str
    severity: str # 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW'
    cvss_score: float # e.g. 9.8
    cvss_vector: str # e.g. 'CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H'
    file_path: str
    line_number: int
    code_snippet: str
    recommendation: str
    patch_diff: Optional[str] = None

class SecurityAgentReportResponse(BaseModel):
    id: uuid.UUID
    repository_id: uuid.UUID
    risk_score: int
    risk_level: str
    critical_count: int
    high_count: int
    medium_count: int
    low_count: int
    findings: List[SecurityFindingItem]
    owasp_distribution: Dict[str, int]
    chart_dataset: Dict[str, Any]
    scanned_at: datetime

    class Config:
        from_attributes = True
